code/modules/overmap/ftl_shunt/computer.dm

```diff
@@ -0,0 +1,150 @@
+/obj/machinery/computer/ship/ftl
+	name = "superluminal shunt control console"
+	icon_keyboard = "teleport_key"
+	icon_screen = "teleport"
+	light_color = "#77fff8"
+	var/obj/machinery/ftl_shunt/core/linked_core
+	var/cost = 0
+
+/obj/machinery/computer/ship/ftl/Initialize()
+	. = ..()
+	find_core()
+
+/obj/machinery/computer/ship/ftl/Destroy()
+	. = ..()
+	if(linked_core)
+		linked_core.ftl_computer = null
+		linked_core = null
+
+/obj/machinery/computer/ship/ftl/proc/recalc_cost()
+	var/jump_dist = get_dist(linked, locate(linked_core.shunt_x, linked_core.shunt_y, GLOB.using_map.overmap_z))
+	var/jump_cost = ((linked.vessel_mass * JOULES_PER_TON) / 1000) * jump_dist
+	return jump_cost
+
+/obj/machinery/computer/ship/ftl/proc/find_core()
+	if(!linked)
+		return
+
+	if(linked_core)
+		linked_core.ftl_computer = null
+		linked_core = null
+
+	for(var/obj/machinery/ftl_shunt/core/C in SSmachines.machinery)
+		if(C.z in linked.map_z)
+			linked_core = C
+			C.ftl_computer = src
+
+/obj/machinery/computer/ship/ftl/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	if(!linked)
+		display_reconnect_dialog(user, "superluminal shunt management system")
+		return
+
+	var/data[0]
+
+	if(!linked_core)
+		find_core()
+		if(!linked_core)
+			to_chat(user, SPAN_WARNING("Unable to establish connection to superluminal shunt."))
+			return
+	recalc_cost()
+
+	data["ftlstatus"] = linked_core.get_status()
+	data["shunt_x"] = linked_core.shunt_x
+	data["shunt_y"] = linked_core.shunt_y
+	data["fuel_joules"] = (linked_core.get_fuel(linked_core.fuel_ports) / 1000)
+	data["jumpcost"] = recalc_cost()
+	data["chargetime"] = round((linked_core.get_charge_time() / 600))
+	data["chargepercent"] = linked_core.chargepercent
+	data["maxfuel"] = linked_core.get_fuel_maximum(linked_core.fuel_ports)
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "ftl_computer.tmpl", "[linked.name] Superluminal Shunt Control", 420, 530, src)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/computer/ship/ftl/OnTopic(var/mob/user, var/list/href_list, state)
+	if(..())
+		return ..()
+
+	if (!linked)
+		return TOPIC_NOACTION
+
+	if(href_list["set_shunt_x"])
```

Comment:
Lots of boilerplate. Maybe something like:
````
if(href_list["set_shunt_x"] || href_list["set_shunt_y"])

	if(!linked_core)
		return TOPIC_NOACTION

	var/input_x = linked_core.shunt_x
	var/input_y = linked_core.shunt_y
	var/fumble = user.skill_check(SKILL_PILOT, SKILL_ADEPT) ? 0 : rand(-2, 2)

	if(href_list["set_shunt_x"])
		input_x = Clamp((input(user, "Enter Destination X Coordinates", "FTL Computer", input_x) as num|null) + fumble, 1, GLOB.using_map.overmap_size - 1)

	if(href_list["set_shunt_y"])
		input_y = Clamp((input(user, "Enter Destination Y Coordinates", "FTL Computer", input_y) as num|null) + fumble, 1, GLOB.using_map.overmap_size - 1)

	if(!CanInteract(user, state))
		return TOPIC_NOACTION
	if(fumble)
		to_chat(user, SPAN_WARNING("You fumble the input because of your inexperience!"))
	linked_core.shunt_x = input_x
	linked_core.shunt_y = input_y
	return TOPIC_REFRESH
````
Haven't even tried to compile this, mind.

Reply:
Implemented.